### PR TITLE
fix: computed register requires at least one limb

### PR DESCRIPTION
**Pull request by DavePearce**
> [!NOTE]
> Moves field-agnostic subdivision into MIR with a new pipeline, updates Concretize API/callers, enhances S‑expression formatting (quoting and numbers), and re-enables several lookup tests.
> 
> - **MIR/Core**:
>   - Centralizes field‑agnostic subdivision with new `Subdivide` pipeline (`mir/subdivide*.go`), covering assignments and constraints; removes legacy subdivision (`mir/agnosticity.go`, `schema/agnostic/divisible.go`).
>   - `Concretize` now takes externs and modules, invokes `Subdivide`; asm concretization updated accordingly.
>   - Adds allocator/constant-register helpers; tweaks vanishing/lookup subdivision and polynomial handling (init zero via `Const64`).
> - **S‑Expression/Printing**:
>   - `sexp.Formatter` gains quoting flag; introduce `sexp.Number`; constants now emit numbers; update printers to `NewFormatter(width, true)`; improve symbol quoting rules.
> - **IR Cleanup**:
>   - Drop per-assignment `Subdivide` methods; move/refactor utilities (e.g., `SubdivideRegisterRefs`).
>   - Fix limb splitting edge case in `register.SplitIntoLimbs` (`<= 1`).
> - **Tests**:
>   - Re-enable and expand lookup tests; broaden field configs in several tests.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c8b1507881d4c68865b8ce29c316826f42607fdd. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: pkg/asm/concretize.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func Concretize[F Element[F]](cfg field.Config, hp MicroHirProgram,
 		modules[i] = ir.BuildModule[F, mir.Constraint[F], mir.Term[F], mir.Module[F]](*m.Module)
 	}
 	// Concretize legacy components
-	copy(modules[n:], mir.Concretize[word.BigEndian, F](mapping, uint(n), p.Externs()))
+	copy(modules[n:], mir.Concretize[word.BigEndian, F](mapping, modules[:n], p.Externs()))
```
[the remark this answers is not in the document]
Comment (DavePearce) [reply pkg/asm/concretize.go]: Disagree
